examples/daal/cpp_sycl/source/logistic_regression/log_reg_dense_online.cpp

```diff
@@ -0,0 +1,143 @@
+
+        auto model = trainModel();
+        printNumericTable(model->getBeta(), "Logistic Regression coefficients:");
+
+        testModel(model);
+    }
+
+    return 0;
+}
+
+logistic_regression::ModelPtr trainModel()
+{
+    constexpr bool isIntercept           = false;
+    constexpr float l2Penalty            = 0.0f;
+    constexpr float learningRate         = 1.0f;
+    constexpr size_t nEpochs             = 20;
+    constexpr size_t nIterationsPerBlock = 5;
+    constexpr size_t dataBlockRowCount   = 500;
+
+    auto xBlock = SyclHomogenNumericTable<>::create(nFeatures, 0, NumericTable::notAllocate);
+    auto yBlock = SyclHomogenNumericTable<>::create(1, 0, NumericTable::notAllocate);
+    NumericTablePtr mergedDataBlock(new MergedNumericTable(xBlock, yBlock));
+
+    LogisticRegressionOnline alg;
+    alg.setParams(nClasses, nFeatures, isIntercept, l2Penalty, dataBlockRowCount, nIterationsPerBlock);
+
+    for (size_t epoch = 0; epoch < nEpochs; epoch++)
+    {
+        FileDataSource<CSVFeatureManager> dataSource(trainDatasetFileName,
+                                                    DataSource::notAllocateNumericTable,
+                                                    DataSource::doDictionaryFromContext);
+
+        while (dataSource.loadDataBlock(dataBlockRowCount, mergedDataBlock.get()) == dataBlockRowCount)
+        {
+            alg.setIterationParams(learningRate);
+            alg.setInput(xBlock, yBlock);
+            alg.compute();
+        }
+    }
+
+    alg.finalizeCompute();
+    return alg.getModel();
+}
+
+void testModel(const logistic_regression::ModelPtr& model)
+{
+    namespace class_prediction = classifier::prediction;
+    using LogRegPrediction = logistic_regression::prediction::Batch<>;
+
+    FileDataSource<CSVFeatureManager> dataSource(testDatasetFileName,
+                                                 DataSource::notAllocateNumericTable,
+                                                 DataSource::doDictionaryFromContext);
+
+    /* Create Numeric Tables for training data and dependent variables */
+    auto testX = SyclHomogenNumericTable<>::create(nFeatures, 0, NumericTable::notAllocate);
+    auto testY = SyclHomogenNumericTable<>::create(1, 0, NumericTable::notAllocate);
+    NumericTablePtr mergedData(new MergedNumericTable(testX, testY));
+
+    /* Retrieve the data from input file */
+    dataSource.loadDataBlock(mergedData.get());
+
+    /* Create an algorithm object to predict values of logistic regression */
+    LogRegPrediction algorithm(nClasses);
+    algorithm.parameter().resultsToEvaluate |= classifier::computeClassProbabilities | classifier::computeClassLogProbabilities;
+
+    /* Pass a testing data set and the trained model to the algorithm */
+    algorithm.input.set(class_prediction::data, testX);
+    algorithm.input.set(class_prediction::model, model);
+
+    /* Predict values of logistic regression */
+    algorithm.compute();
```

Comment:
[minor] prediction would also be good to show online, especially it's very easy to follow.

Reply:
It's done this way for simplicity - if you insist on the change, I can apply it